src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -175,7 +177,8 @@ class GTGRunner(val masterURL: String,
         def updateComponent(label: Double, frame: String, value: Double, row: Int, col: Int): Unit = {
           if (label != 0.0) {
             val node = nodeMap.getOrElse(frame + ":" + label, new MCCNode(frame, label))
-            nodeMap(frame + ":" + label) = node.update(value, row, col)
+            node.updateNodeData(value, row, col)
```

Comment:
Great that you caught this and renamed the function!
updateNodeData is definitely better than update, since update in scala allows you to have syntatic sugar with the = sign.